zoo/analytics/tasks/utils.py

```diff
@@ -58,3 +61,250 @@ def resolve_unknown_format(version: str) -> str:
         else:
             break
     return ".".join(temp_version)
+
+
+class DockerImageId:
+    """Represents parsed docker image id.
+
+    TODO: move this to some more appropriate OSS package.
+    """
+
+    # pylint: disable=too-many-instance-attributes # Ignore PyCommentedCodeBear
+
+    def __init__(self, image: str):  # Ignore RadonBear
+        self.__full_image_id = " ".join(image.split())
+
+        self.__registry = None
+        self.__registry_host_name = None
+        self.__registry_port = None
+        self.__full_image_name = None
+        self.__full_image_name_no_tag = None
+        self.__username = None
+        self.__full_repository_name = None
+        self.__repository_name = None
+        self.__namespaces = None
+        self.__tag = None
+        self.__version = None
```

Comment:
I'd probably save this as `latest`, what do you think?

Reply:
I think it's better to leave as `None` and particularize for each thing when needed